todo/src/reducer/index.js

```diff
@@ -0,0 +1,39 @@
+import { ADD_TODO } from '../action';
+import { TOGGLE_COMPLETE } from '../action';
+
+const initialState = {
+  todos: [
+    {
+      value: 'Walk the dog.',
+      completed: false,
+      id: 0,
+    },
+  ],
+};
+
+export function reducer(state = initialState, action) {
+  switch (action.type) {
+    case ADD_TODO:
+      return {
+        ...state,
+        todos: [...state.todos, action.payload],
+      };
+
+    case TOGGLE_COMPLETE:
+      return {
+        ...state,
+        todos: state.todos.map((todo) => {
+          if (todo === action.payload) {
```

Comment:
You could also have passed only the `todo.id` to the action creator -> reducer, but the outcome would still pretty much be the same.